src/Nethermind/Nethermind.BeaconNode.Peering/PeerSyncStatus.cs

```diff
@@ -0,0 +1,48 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using Nethermind.Core2.Types;
+
+namespace Nethermind.BeaconNode.Peering
+{
+    public class PeerSyncStatus
+    {
+        private readonly object _highestPeerLock = new object();
+
+        public Slot HighestPeerSlot { get; private set; }
+
+        public Slot SyncStartingSlot { get; private set; }
+
+        public void StartSync(Slot slot)
+        {
+            SyncStartingSlot = slot;
+        }
+
+        public void UpdateMostRecentSlot(Slot slot)
+        {
+            if (slot > HighestPeerSlot)
```

Comment:
just wandering if there is anyway to do compare exchange or similar here

Reply:
Done. It is only a single underlying ulong field, so implemented an interlocked compare exchange for slot. Interlocked doesn't yet support unsigned, so had to unsafe cast the ref as Int64.